packages/i18n/src/create-i18n.js

```diff
@@ -149,22 +154,101 @@ export const createI18n = ( initialData, initialDomain ) => {
 
 	/** @type {__} */
 	const __ = ( text, domain ) => {
-		return dcnpgettext( domain, undefined, text );
+		const translation = dcnpgettext( domain, undefined, text );
+		/**
+		 * Filters text with its translation.
+		 *
+		 * @param {string} translation Translated text.
+		 * @param {string} text        Text to translate.
+		 * @param {string} domain      Text domain. Unique identifier for retrieving translated strings.
+		 */
+		return String(
+			applyFilters( 'i18n.gettext', translation, text, domain )
```

Comment:
Core recently added also domain-specific filters named `i18n.gettext_${ domain }`:

https://core.trac.wordpress.org/ticket/49518

Can we add them here, too, to keep the APIs as similar as possible?

Reply:
@jsnajdr I'm happy to update to include that, however there is a slight difference here in implementations. In core "domain" is defaulted to "default" if not provided, but the JS methods don't do that (leaving "domain" as undefined). Given your comments previously about re-use of the resulting object I'm not sure whether we should:
a) update the `__()` etc. method signatures to set domain to "default" if not provided,
b) call a "default" filter if `typeof domain === 'undefined'`

Any thoughts?

Reply:
@leewillis77 The b) option looks better to me as it achieves the goal without changing the library too much. But I'm not really sure. I noticed the new `gettext_${ domain }` hooks when comparing the hooks in this PRs to what's already in Core/PHP. Anyway, adding these extra hooks is not a blocker and can be done at any later time.